### PR TITLE
Disable json support in `read` builtin

### PR DESCRIPTION
**Pull request by siteshwar**
Related: #820

### CODE REVIEW

**Comment by krader1961** [comment]
If we're going to keep the code I'd prefer to introduce a build time flag like we did with `audit-file` in *meson_options.txt* and predicate the json stuff on that symbol. On the other hand I've changed my mind and decided that given that this feature is badly broken for reading and printing JSON and adds a lot of complexity to the "namval" core code I'd just as soon rip it out.

**Comment by siteshwar** [comment]
@krader1961 I don't want to add another `meson` configuration option for it. If people want to enable `json` support, they can do it by defining `SUPPORT_JSON` macro while building.

File: src/cmd/ksh93/bltins/read.c [hunk 1]
```diff
@@ -196,8 +199,13 @@ static_fn int json2sh(Shell_t *shp, Sfio_t *in, Sfio_t *out) {
     }
     return 0;
 }
+#endif
 
-static struct Method methods[] = {{"json", json2sh}, {"ksh", NULL}, {NULL, NULL}};
+static struct Method methods[] = {
+#if JSON_SUPPORT
```
Comment (krader1961) [inline src/cmd/ksh93/bltins/read.c]: s/JSON_SUPPORT/SUPPORT_JSON/

**Comment by krader1961** [comment]
LGTM modulo the one misspelling.

**Pull request merged**